vm_insnhelper.h

```diff
@@ -122,7 +122,8 @@ enum vm_regan_acttype {
 /**********************************************************/
 
 #define CALL_METHOD(calling, ci, cc) do { \
-    VALUE v = (*(cc)->call)(th, GET_CFP(), (calling), (ci), (cc)); \
+    VALUE v = (*(cc)->call)(th, GET_CFP(), (&calling), (ci), (cc)); \
+    calling.recv = 0; \
     if (v == Qundef) { \
```

Comment:
This was my attempt to avoid some compiler warnings. But you are right, fixed now.